### DISCUSSION TITLE
Is It Possible to Directly Manipulate DTSTART and DTEND Times

### DISCUSSION THREAD

**Discussion by 7enderhead** [ical4j/ical4j#780] [Q&A]
- VEVENTs have DTSTART and DTEND times in UTC (e.g., "20150817T160000Z")
- I want to extend the event's time by subtracting 30 minutes from DTSTART and adding 30 minutes to DTEND
- Is it possible to do this directly with DtStart and DtEnd objects, or do I have to take the long tour via DateTime?

**Comment by benfortuna** [comment]
There is currently no easy way to manipulate the time of a Date property "in-place". With the current API I would suggest something like this might be the easiest way to achieve what you're after:

dtStart.setDate(new Period(dtStart.getDate(), new Dur("-30M")).getStart());
dtEnd.setDate(new Period(dtEnd.getDate(), new Dur("30M")).getEnd());

where dtStart and dtEnd are the start and end properties of the event.

**Comment by 7enderhead** [comment]
Thanks for the fast response!

This will get me on the right track.

The example for dtStart has to be adapted slightly, because a Period may not be constructed with an end date before its start date. So its

```
dtStart.setDate(new Period((new Dur("-30M")).getTime(dtStart.getDate())), dtStart.getDate()).getStart());
```
